flatfs.go

```diff
@@ -364,13 +368,21 @@ func (fs *Datastore) renameAndUpdateDiskUsage(tmpPath, path string) error {
 	// Rename and add new file's diskUsage. If the rename fails,
 	// it will either a) Re-add the size of an existing file, which
 	// was sustracted before b) Add 0 if there is no existing file.
-	err = os.Rename(tmpPath, path)
+	for i := 0; i < RetryAttempts; i++ {
+		err = os.Rename(tmpPath, path)
+		// if there's no error, or the source file doesn't exist, abort.
+		if err == nil || os.IsNotExist(err) {
```

Comment:
Not sure if it's worth going down this rabbit hole of understanding what classes of errors are possible here.
* it looks like you'll get an `EEXIST` error in some cases if you try renaming a file to itself.
* renaming across disks (https://stackoverflow.com/questions/50740902/move-a-file-to-a-different-drive-with-go) can fail in some cases. I wonder what happens when your tmp dir is on a different disk than your main data cache.

Reply:
We're putting the temporary directory _inside_ the datastore directory, so that shouldn't be an issue.

Really, we probably don't even need "is not exist", but we might as well have it.